### ISSUE TITLE
Track recurring donation statistics 

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/impact-graph#1477]
We should be able to query and track the following information related to recurring donations feature: 

- total USD value of all recurring donations between two dates
- total USD value of all recurring donations made in a specific token between two dates
- count of active recurring donations created between two dates
- count of active recurring donations made with a speicfic token created between two dates

This information should be available through https://stats.giveth.io/

**Comment by jainkrati** [comment]
@Meriem-BM pls work on this with guidance from @mohammadranjbarz

**Comment by mohammadranjbarz** [comment]
We should implement a new endpoint in `recurringDonationResolver.ts` with name of `getRecurringDonationStats`

## Input
* **beginDate**: string, required
* **endDate**: string, required
See https://github.com/Giveth/impact-graph/blob/staging/src/resolvers/donationResolver.ts#L262-L293 to know how should work with these dates
* **currency** : string, optional

## Output
* **totalStreamedUsdValue**
The query for summing up `totalUsdStreamed` is all donations that have been created in the time range.
If the currency was in input params, you should add the quality of currency in the query

* **activeRecurringDonationsCount**
Count of recurring donations with status:'active' in time range
If the currency was in input params, you should add the quality of currency in the query


**PS**  Please don't forget to write test cases
For writing test cases look at https://github.com/Giveth/impact-graph/blob/staging/src/resolvers/donationResolver.test.ts#L105-L146 to know how to pass dates as input

@divine-comedian  Please read this spec and correct me if there is something wrong
@Meriem-BM Please tell me if something is not clear to you

**Comment by divine-comedian** [comment]
This looks great Mohammad!

**Comment by Meriem-BM** [comment]
@maryjaf, this issue is ready for test

**Comment by maryjaf** [comment]
Please send  the api/query or other thing that I can use it  for test of this issue @Meriem-BM @mohammadranjbarz

**Comment by Meriem-BM** [comment]
@maryjaf here is the query 

```sql
query (
    $beginDate: String!
    $endDate: String!
    $currency: String
    ) {
      getRecurringDonationStats(
        beginDate: $beginDate
        endDate: $endDate
        currency: $currency
      ) {
        totalStreamedUsdValue,
        activeRecurringDonationsCount,
      }
  }
```

with 2 required variables `beginDate` & `endDate` and other optional `currency`.
Date format `YYYY-MM-DD`.

**Comment by maryjaf** [comment]
I want to check the response of the query from admin bro->recurring donation tab, but I got below error 
please take a look @Meriem-BM 
<img width="1246" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNEhuQkE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--dc68a81f2b1aec02a45aed3b4b6479126d19fd44/image.png" alt="image.png" />

**Comment by Meriem-BM** [comment]
@maryjaf It still under development, right @mohammadranjbarz?

**Comment by Meriem-BM** [comment]
Hey @maryjaf can you check this once again

**Comment by maryjaf** [comment]
> Hey @maryjaf can you check this once again

Is it ready for testing ? it is on code review state @Meriem-BM

**Comment by Meriem-BM** [comment]
> > Hey @maryjaf can you check this once again
> 
> Is it ready for testing ? it is on code review state @Meriem-BM

Yes @maryjaf, forgot to move it

**Comment by maryjaf** [comment]
> * total USD value of all recurring donations made in a specific token between two dates
> * count of active recurring donations made with a speicfic token created between two dates

Thanks @Meriem-BM 
I have this Query , how can I test these items related to specific token? @Meriem-BM 

query (
    $beginDate: String!
    $endDate: String!
    $currency: String
    ) {
      getRecurringDonationStats(
        beginDate: $beginDate
        endDate: $endDate
        currency: $currency
      ) {
        totalStreamedUsdValue,
        activeRecurringDonationsCount,
      }
  }

**Comment by Meriem-BM** [comment]
@maryjaf you can pass the token to currency field (`DAI`, `USDT`...)

**Comment by divine-comedian** [comment]
@Meriem-BM is it possible to merge this feature to production?

**Comment by Meriem-BM** [comment]
Yes, the important part is on prod, just last PR isn't which fixes problem of recurring donations tab on admin panel not listing data, do you want me to merge it?

![Screenshot 2024-07-05 at 13 10 14 (2)](https://github.com/Giveth/impact-graph/assets/59566186/534e38e6-aab6-4a5c-8ba0-5fde44a73e03)

**Comment by Meriem-BM** [comment]
> @Meriem-BM is it possible to merge this feature to production?

Yes, the important part is on prod, just last PR isn't which fixes problem of recurring donations tab on admin panel not listing data, do you want me to merge it?

![Screenshot 2024-07-05 at 13 10 14 (2)](https://github.com/Giveth/impact-graph/assets/59566186/534e38e6-aab6-4a5c-8ba0-5fde44a73e03)

**Comment by divine-comedian** [comment]
> > @Meriem-BM is it possible to merge this feature to production?
> 
> Yes, the important part is on prod, just last PR isn't which fixes problem of recurring donations tab on admin panel not listing data, do you want me to merge it?
> 
> ![Screenshot 2024-07-05 at 13 10 14 (2)](https://private-user-images.githubusercontent.com/59566186/346118750-534e38e6-aab6-4a5c-8ba0-5fde44a73e03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNTk3ODMsIm5iZiI6MTcyMTA1OTQ4MywicGF0aCI6Ii81OTU2NjE4Ni8zNDYxMTg3NTAtNTM0ZTM4ZTYtYWFiNi00YTVjLThiYTAtNWZkZTQ0YTczZTAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDE2MDQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwMDY5NGExMGU2YzkwMzA3NzQyMTBkYzM2MTk1ZGM3NzczMjNjOGYzM2VlNmNhMGU0MWU4NzBjOTczYjQ1MmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Q6mdrlH53mCkH8dLcxnhQRvPFP7Sc9KJ_mAY72CYDss)

Just following up - I can see the PR was merged last week - so this feature is ready to go, correct?

**Comment by Meriem-BM** [comment]
> > > @Meriem-BM is it possible to merge this feature to production?
> 
> > 
> 
> > Yes, the important part is on prod, just last PR isn't which fixes problem of recurring donations tab on admin panel not listing data, do you want me to merge it?
> 
> > 
> 
> > ![Screenshot 2024-07-05 at 13 10 14 (2)](https://private-user-images.githubusercontent.com/59566186/346118750-534e38e6-aab6-4a5c-8ba0-5fde44a73e03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNTk3ODMsIm5iZiI6MTcyMTA1OTQ4MywicGF0aCI6Ii81OTU2NjE4Ni8zNDYxMTg3NTAtNTM0ZTM4ZTYtYWFiNi00YTVjLThiYTAtNWZkZTQ0YTczZTAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDE2MDQ0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwMDY5NGExMGU2YzkwMzA3NzQyMTBkYzM2MTk1ZGM3NzczMjNjOGYzM2VlNmNhMGU0MWU4NzBjOTczYjQ1MmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Q6mdrlH53mCkH8dLcxnhQRvPFP7Sc9KJ_mAY72CYDss)
> 
> 
> 
> Just following up - I can see the PR was merged last week - so this feature is ready to go, correct?

Yes
